pkg/flags/github.go

```diff
@@ -56,4 +61,87 @@ func (g *GitHubFlags) Register(set *cli.FlagSet) {
 		Example: "repository-name",
 		Usage:   "The GitHub repository name.",
 	})
+
+	f.StringVar(&cli.StringVar{
+		Name:   "github-app-id",
+		EnvVar: "GITHUB_APP_ID",
+		Target: &g.FlagGitHubAppID,
+		Usage:  "The ID of GitHub App to use for requesting tokens to make GitHub API calls.",
+	})
+
+	f.StringVar(&cli.StringVar{
+		Name:   "github-app-installation-id",
+		EnvVar: "GITHUB_APP_INSTALLATION_ID",
+		Target: &g.FlagGitHubAppInstallationID,
+		Usage:  "The Installation ID of GitHub App to use for requesting tokens to make GitHub API calls.",
+	})
+
+	f.StringVar(&cli.StringVar{
+		Name:   "github-app-private-key-resource-name",
+		EnvVar: "GITHUB_APP_PRIVATE_KEY_RESOURCE_NAME",
+		Target: &g.FlagGitHubAppPrivateKeyResourceName,
+		Usage:  "The resource name of the private key to use with the GitHub App.",
+	})
+
+	set.AfterParse(func(merr error) error {
+		if g.FlagGitHubToken == "" && g.FlagGitHubAppID == "" {
+			merr = errors.Join(merr, fmt.Errorf("one of github token or github app id are required"))
+		}
+		if g.FlagGitHubToken != "" && g.FlagGitHubAppID != "" {
+			merr = errors.Join(merr, fmt.Errorf("only one of github token or github app id are allowed"))
+		}
+		if g.FlagGitHubAppID != "" && g.FlagGitHubAppInstallationID == "" {
+			merr = errors.Join(merr, fmt.Errorf("a github app installation id is required when using a github app id"))
+		}
+		if g.FlagGitHubAppID != "" && g.FlagGitHubAppPrivateKeyResourceName == "" {
+			merr = errors.Join(merr, fmt.Errorf("a github app private key resource name is required when using a github app id"))
+		}
+		return merr
+	})
+}
+
+func (g *GitHubFlags) GetTokenSource(ctx context.Context, permissions map[string]string) (githubauth.TokenSource, error) {
+	if g.FlagGitHubToken != "" {
+		githubTokenSource, err := githubauth.NewStaticTokenSource(g.FlagGitHubToken)
+		if err != nil {
+			return nil, fmt.Errorf("failed to create github static token source: %w", err)
+		}
+		return githubTokenSource, nil
+	} else {
+		sm, err := secretmanager.NewClient(ctx)
+		if err != nil {
+			return nil, fmt.Errorf("failed to create secret manager client: %w", err)
+		}
+		defer sm.Close()
+
+		privateKeyPEM, err := accessSecret(ctx, sm, g.FlagGitHubAppPrivateKeyResourceName)
+		if err != nil {
+			return nil, fmt.Errorf("failed to get github private key pem: %w", err)
+		}
+
+		app, err := githubauth.NewApp(g.FlagGitHubAppID, g.FlagGitHubAppInstallationID, privateKeyPEM)
+		if err != nil {
+			return nil, fmt.Errorf("failed to create github app token source: %w", err)
+		}
+
+		return app.SelectedReposTokenSource(permissions, g.FlagGitHubRepo), nil
+	}
```

Comment:
Now that we that the call to secret manager has been removed I'm not sure we need change this, but let me know what you think.

Reply:
I still think it's better to frontload the creation of these objects. It will also simplify the callsites, because you only need to check a single error return. 

We can merge this as-is, and I can do a follow-up PR to show you what I mean.